sample-parent/sample-features/conditional-outputs/src/main/java/org/talend/sdk/component/sample/feature/availableoutput/output/AvailableOutputWithAnnotationTwo.java

```diff
@@ -0,0 +1,71 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.talend.sdk.component.sample.feature.availableoutput.output;
+
+import static org.talend.sdk.component.api.component.Icon.IconType.CUSTOM;
+
+import java.io.Serializable;
+
+import javax.annotation.PostConstruct;
+
+import org.talend.sdk.component.api.component.Icon;
+import org.talend.sdk.component.api.component.Version;
+import org.talend.sdk.component.api.configuration.Option;
+import org.talend.sdk.component.api.meta.ConditionalOutput;
+import org.talend.sdk.component.api.meta.Documentation;
+import org.talend.sdk.component.api.processor.ElementListener;
+import org.talend.sdk.component.api.processor.Input;
+import org.talend.sdk.component.api.processor.Output;
+import org.talend.sdk.component.api.processor.OutputEmitter;
+import org.talend.sdk.component.api.processor.Processor;
+import org.talend.sdk.component.api.record.Record;
+import org.talend.sdk.component.sample.feature.availableoutput.service.AvaiableoutputService;
+
+@Version(1)
+// default version is 1, if some configuration changes happen between 2 versions you can add a migrationHandler
+@Icon(value = CUSTOM, custom = "AvailableOutputsOutput")
+// icon is located at src/main/resources/icons/AvailableOutputsOutput.svg
+@Processor(family = "sampleAvailableOutput", name = "Output2")
+@Documentation("Sample for Available output flows.")
+@ConditionalOutput("output-flow2")
+public class AvailableOutputWithAnnotationTwo implements Serializable {
+
+    private final ConfigurationTwo configuration;
+
+    private final AvaiableoutputService service;
+
+    public AvailableOutputWithAnnotationTwo(@Option("configuration") final ConfigurationTwo configuration,
+                                            final AvaiableoutputService service) {
+        this.configuration = configuration;
+        this.service = service;
+    }
+
+    @PostConstruct
+    public void init() {
+        // this method will be executed once for the whole component execution,
+        // this is where you can establish a connection for instance
+        // Note: if you don't need it you can delete it
+    }
+
+    @ElementListener
+    public void process(@Input final Record input,
+            @Output final OutputEmitter<Record> main,
+            @Output(AvaiableoutputService.SECOND_FLOW_NAME2) final OutputEmitter<Record> second,
+            @Output(AvaiableoutputService.THIRD_FLOW_NAME2) final OutputEmitter<Record> third) {
+        //this method here is just used to declare some outputs.
```

Comment:
Maybe for testing purpose, emit some data...

Reply:
test only for right click on the component ...

Reply:
Yes but to be sure it doesn't impact the standard behavior at runtime. Generate some fixed data like:
1,aaa
2,bbb
3,ccc

Reply:
No emit here and marked resolved.

Reply:
No emit here and marked resolved.

Reply:
emit the input. Tested on the job, the job use a fixedInput to give it several records. and it can output them.